### PR TITLE
Fix visu 372

### LINKED ISSUE #372: In pypesto.visualize.sampling_parameters_trace: ValueError: The palette list has the wrong number of colors.
This error occurs frequently since recently.

**Showcase**: https://travis-ci.com/github/ICB-DCM/pyPESTO/builds/168940638

**Error message:**

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-09211946a5d8> in <module>
----> 1 ax = pypesto.visualize.sampling_parameters_trace(result, use_problem_bounds=False, full_trace=True, size=(12,5))
~/virtualenv/python3.8.1/lib/python3.8/site-packages/pypesto/visualize/sampling.py in sampling_parameters_trace(result, i_chain, full_trace, stepsize, use_problem_bounds, suptitle, size, ax)
    163         ax = axes[plot_id]
    164 
--> 165         ax = sns.scatterplot(x="iteration", y=plot_id, data=params_fval,
    166                              ax=ax, **kwargs)
    167 
~/virtualenv/python3.8.1/lib/python3.8/site-packages/seaborn/relational.py in scatterplot(x, y, hue, style, size, data, palette, hue_order, hue_norm, sizes, size_order, size_norm, markers, style_order, x_bins, y_bins, units, estimator, ci, n_boot, alpha, x_jitter, y_jitter, legend, ax, **kwargs)
   1388                 legend="brief", ax=None, **kwargs):
   1389 
-> 1390     p = _ScatterPlotter(
   1391         x=x, y=y, hue=hue, style=style, size=size, data=data,
   1392         palette=palette, hue_order=hue_order, hue_norm=hue_norm,
~/virtualenv/python3.8.1/lib/python3.8/site-packages/seaborn/relational.py in __init__(self, x, y, hue, size, style, data, palette, hue_order, hue_norm, sizes, size_order, size_norm, dashes, markers, style_order, x_bins, y_bins, units, estimator, ci, n_boot, alpha, x_jitter, y_jitter, legend)
    891         )
    892 
--> 893         self.parse_hue(plot_data["hue"], palette, hue_order, hue_norm)
    894         self.parse_size(plot_data["size"], sizes, size_order, size_norm)
    895         self.parse_style(plot_data["style"], markers, None, style_order)
~/virtualenv/python3.8.1/lib/python3.8/site-packages/seaborn/relational.py in parse_hue(self, data, palette, order, norm)
    367             cmap = None
    368             limits = None
--> 369             levels, palette = self.categorical_to_palette(
    370                 # List comprehension here is required to
    371                 # overcome differences in the way pandas
~/virtualenv/python3.8.1/lib/python3.8/site-packages/seaborn/relational.py in categorical_to_palette(self, data, order, palette)
    214                 if len(palette) != n_colors:
    215                     err = "The palette list has the wrong number of colors."
--> 216                     raise ValueError(err)
    217                 colors = palette
    218             else:
ValueError: The palette list has the wrong number of colors.
```

### PR DESCRIPTION
**Pull request by erikadudki**
`Error in: ValueError: The palette list has the wrong number of colors.`
depending on the status of the vector 'converged' (if all solutions converged (->1 color), or a subset converged (->2 colors)), the color vector (-> palette) needed to be adjusted

closes #372

### CODE REVIEW

**Comment by dweindl** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pypesto/visualize/sampling.py  2
         

Clones added
============
- pypesto/visualize/sampling.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/dweindl/pyPESTO/pullRequest?prid=5625307&bid=18427710)

File: pypesto/visualize/sampling.py [hunk 1]
```diff
@@ -61,7 +61,10 @@ def sampling_fval_trace(
               's': 10}
     if full_trace:
         kwargs['hue'] = "converged"
-        kwargs['palette'] = ["#868686", "#477ccd"]
+        if len(params_fval[kwargs['hue']].unique()) == 1:
+            kwargs['palette'] = ["#477ccd"]
+        elif len(params_fval[kwargs['hue']].unique()) == 2:
```
Comment (yannikschaelte) [inline pypesto/visualize/sampling.py:66]: it cannot be more than 2?

Comment (erikadudki) [reply pypesto/visualize/sampling.py:66]: good question, I thought the converged status can only be '0' or '1', but I'm not sure about that, are there other possibilities for the status 'converged'?

Comment (yannikschaelte) [reply pypesto/visualize/sampling.py:66]: converged is a binary indicator for whether this is still burn-in or the chain after burn-in, right? yes, then it should be one or 2.

**Pull request merged**